plugins/events-node/src/service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  coreServices,
+  createServiceFactory,
+  createServiceRef,
+} from '@backstage/backend-plugin-api';
+import { EventService, DefaultEventService } from './api';
+
+/**
+ * The EventService that allows to publish events, and subscribe to topics.
+ * Uses the `root` scope so that events can be shared across all plugins, modules, and more.
+ *
+ * @public
+ */
+export const eventServiceRef = createServiceRef<EventService>({
+  id: 'events.service',
+  scope: 'root',
+});
+
+/**
+ * Factory for the {@link eventServiceRef}.
+ *
+ * @public
+ */
+export const eventServiceFactory = createServiceFactory({
```

Comment:
One option for avoiding the need for an explicit installation of this would be to add a `defaultFactory` to `eventServiceRef`. That has the downside of potentially duplicating the instances though if plugins consume different versions, and since we rely on shared state for this implementation we'd need to work around that with a global.

Tbh I think we can skip all that and just add this to `@backstage/backend-defaults` straight away. Just import this factory directly and add it to [this list](https://github.com/backstage/backstage/blob/7af5f7320f87475a6cc0581105657c295e2516bf/packages/backend-defaults/src/CreateBackend.ts#L37). With that we should avoiding having it as the default export of this package as well, since library packages are not intended to be installed directly.

For alternative implementation we'd do something like `@backstage/plugin-events-backend-pubsub-service` or similar, exact naming pattern is TBD

Reply:
See https://github.com/backstage/backstage/pull/22344#discussion_r1471138170 suggesting doing it as a default factory on the ref instead. We talked about it among ourselves just now (accidentally reviewed the same PR at the same time :)) and think that maybe that's better for now.

Reply:
Yep, adding to `backend-defaults` is a bit more risky from a stability PoV, so best to aim for a default factory instead.